src/modules/control_allocator/ControlAllocator.cpp

```diff
@@ -239,6 +239,7 @@ ControlAllocator::update_effectiveness_source()
 			break;
 
 		case EffectivenessSource::ROVER_DIFFERENTIAL:
+			tmp = new ActuatorEffectivenessRoverDifferential();
```

Comment:
I would not bring back the differential rover control allocation but rather have a differential boat `CA_AIRFRAME` and directly publish to the actuator functions like it's done for differential rovers here: https://github.com/PX4/PX4-Autopilot/blob/8ceeda730d12583a8709b54f07b7c9ffbe67a9b8/src/modules/differential_drive/DifferentialDriveKinematics/DifferentialDriveKinematics.cpp#L63-L67

Reply:
Thank you for your comment @MaEtUgR ! I used the differential rover control allocation because the control allocation is exactly the same for my boat and it's pretty convenient to only deal with thrust and torque. 
Why the control allocation is not used for the differential rover ? Is it to ensure an accurate posisitionning ? 

I can reuse the differential rover control allocation and create a new CA_AIRFRAME for the differential boat if it's fine for you.